cmd/chisel/cmd_info_test.go

```diff
@@ -0,0 +1,260 @@
+	stdout: `
+		package: mypkg1
+		archive: ubuntu
+		slices:
+			myslice1:
+				contents:
+					/dir/file: {}
+					/dir/sub-dir/: {make: true, mode: 0644}
+	`,
+	err: `no slice definitions found for: "foo", "bar_foo"`,
+}, {
+	summary: "No args",
+	input:   infoRelease,
+	err:     "the required argument `<pkg|slice> (at least 1 argument)` was not provided",
+}, {
+	summary: "Empty, whitespace args",
+	input:   infoRelease,
+	query:   []string{"", "    "},
+	err:     `no slice definitions found for: "", "    "`,
+}, {
+	summary: "Ignore invalid slice names",
+	input:   infoRelease,
+	query:   []string{"foo_bar_foo", "a_b", "7_c", "a_b c", "a_b x_y"},
+	err:     `no slice definitions found for: "foo_bar_foo", "a_b", "7_c", "a_b c", "a_b x_y"`,
+}}
+
+var testKey = testutil.PGPKeys["key1"]
+var defaultChiselYaml = `
+	format: chisel-v1
+	archives:
+		ubuntu:
+			version: 22.04
+			components: [main, universe]
+			v1-public-keys: [test-key]
+	v1-public-keys:
+		test-key:
+			id: ` + testKey.ID + `
+			armor: |` + "\n" + testutil.PrefixEachLine(testKey.PubKeyArmor, "\t\t\t\t\t\t") + `
+`
+
+var infoRelease = map[string]string{
+	"chisel.yaml": string(defaultChiselYaml),
+	"slices/mypkg1.yaml": `
+		package: mypkg1
+		essential:
+			- mypkg1_myslice1
+		slices:
+			myslice1:
+				contents:
+					/dir/file:
+			myslice2:
+				essential:
+					- mypkg2_myslice
+	`,
+	"slices/mypkg2.yaml": `
+		package: mypkg2
+		slices:
+			myslice:
+				contents:
+					/dir/another-file:
+	`,
+	"slices/mypkg3.yaml": `
+		package: mypkg3
+		essential:
+			- mypkg1_myslice1
+		slices:
+			myslice:
+				essential:
+					- mypkg2_myslice
+				contents:
+					/dir/other-file:
+					/dir/glob*:
+					/dir/sub-dir/:       {make: true, mode: 0644}
+					/dir/copy:           {copy: /dir/file}
+					/dir/symlink:        {symlink: /dir/file}
+					/dir/mutable:        {text: TODO, mutable: true, arch: riscv64}
+					/dir/arch-specific*: {arch: [amd64,arm64,i386]}
+					/dir/until:          {until: mutate}
+					/dir/unfolded:
+						copy: /dir/file
+						mode: 0644
+				mutate: |
+					# Test multi-line string.
+					content.write("/dir/mutable", foo)
+	`,
+}
+
+func (s *ChiselSuite) TestInfoCommand(c *C) {
+	for _, test := range infoTests {
+		c.Logf("Summary: %s", test.summary)
+
+		s.ResetStdStreams()
+
+		dir := c.MkDir()
+		for path, data := range test.input {
+			fpath := filepath.Join(dir, path)
+			err := os.MkdirAll(filepath.Dir(fpath), 0755)
+			c.Assert(err, IsNil)
+			err = os.WriteFile(fpath, testutil.Reindent(data), 0644)
+			c.Assert(err, IsNil)
+		}
+		test.query = append([]string{"info", "--release", dir}, test.query...)
+
+		_, err := chisel.Parser().ParseArgs(test.query)
+		if test.err != "" {
+			c.Assert(err, ErrorMatches, test.err)
+			continue
+		}
+		c.Assert(err, IsNil)
+		test.stdout = string(testutil.Reindent(test.stdout))
+		c.Assert(strings.TrimSpace(s.Stdout()), Equals, strings.TrimSpace(test.stdout))
```

Comment:
What about:
```go
c.Assert(s.Stdout(), Equals, strings.TrimSpace(test.stdout)+"\n")
```
instead. I think the whitespace here is significant because the output of `chisel find` is printed to the terminal. We can ensure that it ends on a single newline and does not have an empty one at the beginning.

Reply:
Done.